java/code/src/com/suse/manager/xmlrpc/admin/AdminPaygHandler.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2021 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.manager.xmlrpc.admin;
+
+import com.redhat.rhn.domain.cloudpayg.PaygSshData;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.frontend.xmlrpc.BaseHandler;
+import com.redhat.rhn.taskomatic.TaskomaticApi;
+
+import com.suse.manager.admin.PaygAdminManager;
+
+import org.apache.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Payg-as-you-go XMLRPC Handler
+ *
+ * @xmlrpc.namespace admin
+ * @xmlrpc.doc Provides methods to access and modify pay-as-you-go ssh connection data
+ */
+public class AdminPaygHandler extends BaseHandler {
+
+    private static final Logger LOG = Logger.getLogger(AdminPaygHandler.class);
+
+    private PaygAdminManager paygAdminManager;
+
+    /**
+     * Constructor with an instance of taskomatic API to be used
+     * @param taskomaticApi taskomatic api instance
+     */
+    public AdminPaygHandler(TaskomaticApi taskomaticApi) {
+        this.paygAdminManager = new PaygAdminManager(taskomaticApi);
+    }
+
+    /**
+     *
+     * @param loggedInUser The current user
+     * @param description Description for the ssh connection data
+     * @param host hostname or IP address to the instance
+     * @param port shh port to the instance
+     * @param username ssh username to connection on the instance
+     * @param password ssh password for the instance
+     * @param key private key to use in authentication
+     * @param keyPassword private key password
+     * @return Returns 1 if successful (exception otherwise)
+     *
+     * @xmlrpc.doc Create a new ssh connection data to extract data from
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "description")
+     * @xmlrpc.param #param_desc("string", "host", "hostname or IP address to the instance, will fail if
+     * already in use.")
+     * @xmlrpc.param #param("int", "port")
+     * @xmlrpc.param #param("string", "username")
+     * @xmlrpc.param #param("string", "password")
+     * @xmlrpc.param #param_desc("string", "key", "private key to use in authentication")
+     * @xmlrpc.param #param("string", "keyPassword")
+     * @xmlrpc.returntype #return_int_success()
+     */
+    public int create(User loggedInUser, String description, String host, Integer port, String username,
```

Comment:
It'd be good to qualify what `create` creates in the method name I believe.

The full call will be `admin.payg.create(...)` - I have nothing against the `admin` prefix, but it's not like the user is "creating a pay-as-you-go" here. Rather, they are creating an SSH connection to harvest data from a pay-as-you-go host.

Reply:
The name `admin.payg.creatSshConnection` would make more sense to you?